raft_paper_test.go

```diff
@@ -832,17 +793,18 @@ func TestVoter(t *testing.T) {
 		wreject bool
 	}{
 		// same logterm
-		{[]pb.Entry{{Term: 1, Index: 1}}, 1, 1, false},
-		{[]pb.Entry{{Term: 1, Index: 1}}, 1, 2, false},
-		{[]pb.Entry{{Term: 1, Index: 1}, {Term: 1, Index: 2}}, 1, 1, true},
+		{index(1).terms(1), 1, 1, false},
+		{index(1).terms(1), 1, 2, false},
+		{index(1).terms(1, 1), 1, 1, true},
 		// candidate higher logterm
-		{[]pb.Entry{{Term: 1, Index: 1}}, 2, 1, false},
-		{[]pb.Entry{{Term: 1, Index: 1}}, 2, 2, false},
+		{index(1).terms(1), 2, 1, false},
+		{index(1).terms(1), 2, 2, false},
 		{[]pb.Entry{{Term: 1, Index: 1}, {Term: 1, Index: 2}}, 2, 1, false},
 		// voter higher logterm
-		{[]pb.Entry{{Term: 2, Index: 1}}, 1, 1, true},
-		{[]pb.Entry{{Term: 2, Index: 1}}, 1, 2, true},
-		{[]pb.Entry{{Term: 2, Index: 1}, {Term: 1, Index: 2}}, 1, 1, true},
```

Comment:
This line was incorrect: terms in a log can not regress. Added two correct tests instead.